testdata/unchecked-type-assertion.go

```diff
@@ -0,0 +1,102 @@
+package fixtures
+
+import "fmt"
+
+var (
+	foo  any = "foo"
+	bars     = []any{1, 42, "some", "thing"}
+)
+
+func handleIgnored() {
+	r, _ := foo.(int) // MATCH /type cast result is unchecked in foo.(int) - type assertion result ignored/
+}
+
+func handleSkipped() {
+	r := foo.(int) // MATCH /type cast result is unchecked in foo.(int) - type assertion will panic if not matched/
+}
+
+func handleReturn() int {
+	return foo.(int) // MATCH /type cast result is unchecked in foo.(int) - type assertion will panic if not matched/
+}
+
+func handleSwitch() {
+	switch foo.(int) { // MATCH /type cast result is unchecked in foo.(int) - type assertion will panic if not matched/
+	case 0:
+	case 1:
+		//
+	}
+}
+
+func handleRange() {
+	var some any = bars
+	for _, x := range some.([]string) { // MATCH /type cast result is unchecked in some.([]string) - type assertion will panic if not matched/
+		fmt.Println(x)
+	}
+}
+
+func handleTypeSwitch() {
+	// Should not be a lint
+	switch foo.(type) {
+	case string:
+	case int:
+		//
+	}
+}
+
+func handleTypeSwitchWithAssignment() {
+	// Should not be a lint
+	switch n := foo.(type) {
+	case string:
+	case int:
+		//
+	}
+}
+
+func handleTypeComparison() {
+	if foo.(int) == 1 { // MATCH /type cast result is unchecked in foo.(int) - type assertion will panic if not matched/
+		return
+	}
+}
+
+func handleTypeComparisonReverse() {
+	if foo.(int) == 1 { // MATCH /type cast result is unchecked in foo.(int) - type assertion will panic if not matched/
```

Comment:
I think, it's a small mistake here, it had to be:

```suggestion
	if 1 == foo.(int) { // MATCH /type cast result is unchecked in foo.(int) - type assertion will panic if not matched/
```